### PR TITLE
[fix][transaction] Fix some tests and clean up logs

### PR DESCRIPTION
**Pull request by gaoran10**
### Modifications

Fix some tests.

### Documentation

Check the box below.

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/storage/PartitionLog.java [hunk 1]
```diff
@@ -1102,15 +1117,17 @@ public CompletableFuture<Long> fetchOldestAvailableIndexFromTopic() {
...
+        long numberOfEntries = managedLedger.getNumberOfEntries();
+        if (numberOfEntries == 0) {
             long currentOffset = MessageMetadataUtils.getCurrentOffset(managedLedger);
             log.info("First offset for topic {} is {} as the topic is empty (numberOfEntries=0)",
                     fullPartitionName, currentOffset);
             future.complete(currentOffset);
 
             return future;
         }
+        log.info("{} numberOfEntries={}", fullPartitionName, numberOfEntries);
```
Comment (gaoran10) [inline kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/storage/PartitionLog.java:1130]: Maybe need to remove.

**Comment by Demogorgon314** [comment]
@gaoran10 Please help resolve conflicts.

File: kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/storage/PartitionLog.java [hunk 2]
```diff
@@ -192,6 +195,10 @@ public boolean isInitialisationFailed() {
         return initFuture.isDone() && initFuture.isCompletedExceptionally();
     }
 
+    public void markAsUnloaded() {
```
Comment (Demogorgon314) [inline kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/storage/PartitionLog.java:198]: Is there a unit test available for this?

Comment (gaoran10) [reply kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/storage/PartitionLog.java:198]: Please check [here](https://github.com/streamnative/kop/pull/1966/files#diff-f7a7a4e368793ba72976ed229fe196eae7d34eede34240e28a8e18a0ac3e04f1R1450)

**Review by Demogorgon314 — APPROVED** [review]

**Pull request merged**